ql/src/java/org/apache/hadoop/hive/ql/Driver.java

```diff
@@ -1058,15 +811,15 @@ private int closeInProcess(boolean destroyed) {
     releaseResStream();
     releaseContext();
     if (destroyed) {
-      if (!hiveLocks.isEmpty()) {
+      if (!driverTxnHandler.hasHiveLock()) {
```

Comment:
Again, maybe just a driverTxnHandler.release?

Reply:
I agree, I'll modify this.

Reply:
I agree, I'll modify this.